atlas_anndata/anndata_ops.py

```diff
@@ -242,47 +239,52 @@ def derive_metadata(adata, config, kind=None):
                     adata.obs[sample_field], adata.obs_names
                 )
             ]
-
         else:
             print("...deriving runs and barcodes by parsing cell IDs")
             runs, barcodes = parse_cell_ids(adata)
 
             # If we had to derive run IDs from cell IDs, save them in the
             # metadata, including in the original anndata object, so they can
-            # beused for make cell/ library mappings
+            # be used for make cell/ library mappings
             cell_metadata[sample_field] = adata.obs[sample_field] = runs
 
+        if sample_field not in cell_metadata.columns:
+            cell_metadata[sample_field] = adata.obs[sample_field]
+
```

Comment:
I wonder if this conditional is redundant with the previous `else` statement, since `cell_metadata` is also from `adata.obs`?  Or is there a case where `sample_field` is present in `adata.obs.columns` but not in `cell_metadata.columns`, e.g., if it is possible for `sample_field` to be filtered out because it is not the correct `slot_def["kind"]`?

Reply:
Thanks Iris for the review. But this is not redundant.

When we initialise the bundle, it first generate the config yaml file, and then run derive_metadata. At this time, MISSING in sample_field is True, and it will enter the else condition above. The "sample" column will be included in the .obs, but not be included in the config yaml file (as it is generated before the "sample" column is derived). This makes that when we initialise the magetab files, the "sample" column is not existed in cell_metadata. This is because columns in cell_metadata are chosen according to the definition of fields in the config yaml file. The field "sample" is not in the config yaml file, so it will be not included in cell_metadata anyway. Therefore, we need to add it in cell_metadata if it's not there.